Fix broken Neo4j installation

jsf9k opened the pull request:
## 🗣 Description ##

This pull request fixes the broken Neo4j installation.

## 💭 Motivation and context ##

Resolves cisagov/cool-system-internal#124.

## 🧪 Testing ##

All automated tests pass.  I also built a new COOL staging AMI with these changes and verified that it functions as expected.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.
- [x] Bump version

## ✅ Pre-merge checklist ##

- [x] Revert to the default branch once cisagov/ansible-role-openjdk#41 is merged.
- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

jsf9k committed:
Upgrade to version 4.4 of Neo4j

jsf9k committed:
Add code to configure Debian to prefer OpenJDK 11 vice 17

By default OpenJDK 17 is being used on Kali Linux, although OpenJDK 11
is also installed.  Neo4j <5 requires OpenJDK 11.

jsf9k committed:
Add appropriate references to issue for upgrading to Neo4j 5 when possible

jsf9k committed:
Ensure that OpenJDK 11 is installed before installing Neo4j

Neo4j <5 requires OpenJDK 11, but Kali Linux is currently using
OpenJDK 17 by default.

jsf9k committed:
Bump version from 0.5.11 to 0.5.12

jsf9k committed:
Bump version from 0.5.12 to 0.5.12-rc.1

mcdonnnj commented on src/bloodhound.yml:
```diff
@@ -1,17 +1,34 @@
+        - name: Install OpenJDK 11
+          ansible.builtin.import_role:
+            name: openjdk
+        - name: Configure Debian to use OpenJDK 11
+          community.general.alternatives:
+            name: java
+            path: /usr/lib/jvm/java-11-openjdk-amd64/bin/java
```
Do we need to worry about any of the other common Java commands (`javac` and `javaw` off the top of my head)?

jsf9k replied:
I don't think so, since we are only running Java code (Neo4j and Cobalt Strike) but here is the complete list for OpenJDK 11 (from `update-alternatives --get-selections`):
```
jaotc                          auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jaotc
jar                            auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jar
jarsigner                      auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jarsigner
java                           auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/java
javac                          auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/javac
javadoc                        auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/javadoc
javap                          auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/javap
jcmd                           auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jcmd
jconsole                       auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jconsole
jdb                            auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jdb
jdeprscan                      auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jdeprscan
jdeps                          auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jdeps
jexec                          auto     /usr/lib/jvm/java-11-openjdk-amd64/lib/jexec
jfr                            auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jfr
jhsdb                          auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jhsdb
jimage                         auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jimage
jinfo                          auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jinfo
jjs                            auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jjs
jlink                          auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jlink
jmap                           auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jmap
jmod                           auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jmod
jps                            auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jps
jrunscript                     auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jrunscript
jshell                         auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jshell
jstack                         auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jstack
jstat                          auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jstat
jstatd                         auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/jstatd
keytool                        auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/keytool
pack200                        auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/pack200
rmic                           auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/rmic
rmid                           auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/rmid
rmiregistry                    auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/rmiregistry
serialver                      auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/serialver
unpack200                      auto     /usr/lib/jvm/java-11-openjdk-amd64/bin/unpack200
```

It's worth noting that the list of programs is slightly different for OpenJDK 17:
```
jar                            auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jar
jarsigner                      auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jarsigner
java                           auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/java
javac                          auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/javac
javadoc                        auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/javadoc
javap                          auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/javap
jcmd                           auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jcmd
jconsole                       auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jconsole
jdb                            auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jdb
jdeprscan                      auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jdeprscan
jdeps                          auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jdeps
jexec                          auto     /usr/lib/jvm/java-17-openjdk-amd64/lib/jexec
jfr                            auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jfr
jhsdb                          auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jhsdb
jimage                         auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jimage
jinfo                          auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jinfo
jlink                          auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jlink
jmap                           auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jmap
jmod                           auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jmod
jpackage                       auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jpackage
jps                            auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jps
jrunscript                     auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jrunscript
jshell                         auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jshell
jstack                         auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jstack
jstat                          auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jstat
jstatd                         auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/jstatd
keytool                        auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/keytool
rmiregistry                    auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/rmiregistry
serialver                      auto     /usr/lib/jvm/java-17-openjdk-amd64/bin/serialver
```

jsf9k replied:
In any event, whatever `community.general.alternatives` jazz we decide is required should probably be moved to [cisagov/ansible-role-openjdk](https://github.com/cisagov/ansible-role-openjdk).

jsf9k replied:
Please see cisagov/ansible-role-openjdk#41.

mcdonnnj reviewed (approved):
I have one minor question for you but otherwise LGTM ✔

jsf9k committed:
Temporarily use a non-default branch of cisagov/ansible-role-openjdk

This change is being made for testing purposes, and it can be reverted
after cisagov/ansible-role-openjdk#41 is merged.

jsf9k committed:
Bump version from 0.5.12-rc.1 to 0.5.12-rc.2

jsf9k committed:
Remove alternatives configuration code

This is now handled in cisagov/ansible-role-openjdk.  Please see
cisagov/ansible-role-openjdk#41 for more details.

jsf9k committed:
Revert "Temporarily use a non-default branch of cisagov/ansible-role-openjdk"

This reverts commit 5349cb4150c66a58ac46660c7721b7136425b274.

The non-default branch of cisagov/ansible-role-openjdk that we were
using was merged as part of cisagov/ansible-role-openjdk#41.

dav3r reviewed (approved):
Looks good! 👍

jsf9k committed:
Finalize version from 0.5.12-rc.2 to 0.5.12
